hazelcast/src/main/resources/hazelcast-config-3.8.xsd

```diff
@@ -1172,6 +1173,23 @@
             </xs:simpleType>
         </xs:attribute>
     </xs:complexType>
+
+    <xs:complexType name="lock">
```

Comment:
I've added lock config to config xml generator and spring config. Also fixed queue and ringbuffer config in xml generator.